dcpy/lifecycle/validate/pandera_utils.py

```diff
@@ -0,0 +1,119 @@
+import pandera as pa
+import pandas as pd
+import geopandas as gpd
+from inspect import (
+    signature,
+)  # used for checking expected attributes in a class signuture
+
+from dcpy.models.dataset import Column, CheckAttributes
+
+
+def create_check(check: str | dict[str, CheckAttributes]) -> pa.Check:
+    """
+    Creates a Pandera `Check` object from a given check definition.
+
+    Args:
+        check:
+            A string representing the name of the check or a dictionary with the
+            check name as the key and a dictionary of attributes as the value.
+    Returns:
+        pa.Check:
+            A Pandera `Check` object constructed with the specified parameters.
+    Raises:
+        AssertionError:
+            If the `check` dictionary does not contain exactly one key-value pair.
+        ValueError:
+            If the check name is not registered or if attributes cannot be parsed
+            or used to create a valid `Check`.
+    """
+    allowed_check_names = {
+        **pa.Check.CHECK_FUNCTION_REGISTRY,
+        **pa.Check.REGISTERED_CUSTOM_CHECKS,
+    }
+
+    if isinstance(check, str):
+        check_name = check
+        check_args = None
+    elif isinstance(check, dict):
+        assert (
+            len(check) == 1
+        ), "`utils.create_pa_check` expects exactly 1 key-value pair in `check` param."
+        check_name, check_args = next(iter(check.items()))
+
+    if check_name not in allowed_check_names:
+        raise ValueError(f"Unregistered check name: '{check_name}'.")
+
+    pandera_registered_check = getattr(pa.Check, check_name)
+
+    if check_args:
+        check_expected_params = signature(pandera_registered_check).parameters
+        invalid_check_keys = set(check_args.args.keys()) - set(
+            check_expected_params.keys()
+        )
+        if invalid_check_keys:
```

Comment:
So looking at the Pandera [source](https://github.com/unionai-oss/pandera/blob/813420db5092a7ad4a4ce338b18cf56b1277fbba/pandera/api/checks.py#L294) for this, and I think the intent is to be able to pass **kwargs through the `cls.from_builtin_check_name` pa.Check constructor. I'm guessing that comes into play with custom-registered checks. Which maybe in the long-run means we shouldn't do this type-checking. For now, I think it's fine.

Reply:
Resolving. Chatted in person, and basically you can pass these kwargs either to the built-in fn constructor (which we're disallowing here) or to the Check constructor, which is 👍